### PR TITLE
feat: Improve logging for index operations

### PR DESCRIPTION
**Pull request by hectoras**
**Associated Jira issue:** [ADF-568](https://oat-sa.atlassian.net/browse/ADF-568)

Related merge requests:

- https://github.com/oat-sa/tao-core/pull/3424

### CODE REVIEW

File: model/Index/Service/SyncResultIndexer.php [hunk 1]
```diff
@@ -38,16 +39,35 @@ public function setNormalizer(NormalizerInterface $normalizer): void
         $this->normalizer = $normalizer;
     }
 
+    /**
+     * @throws Throwable
+     * @throws \common_Exception
```
Comment (gabrielfs7) [inline model/Index/Service/SyncResultIndexer.php:44]: FQN

**Review by gabrielfs7 — APPROVED** [review]

File: model/Index/Service/SyncResultIndexer.php (hunk 1 above)
Comment (hectoras) [reply model/Index/Service/SyncResultIndexer.php]: Fixed :point_right:  e353134e362049d3968f689c826ed07de1141bdf :ok_hand:

File: tests/Unit/Index/Service/ResultIndexerTest.php [hunk 2]
```diff
@@ -59,6 +60,7 @@ public function setUp(): void
                 [
                     QueueDispatcherInterface::SERVICE_ID => $this->queueDispatcher,
                     AdvancedSearchChecker::class => $this->advancedSearchChecker,
+                    'generis/log' => $this->createMock(LoggerService::class)
```
Comment (gabrielfs7) [inline tests/Unit/Index/Service/ResultIndexerTest.php:63]: Lets use LoggerService::SERVICE_ID instead.

**Review by gabrielfs7 — APPROVED** [review]
Good job! Please, do not forget to check the unit tests :)

File: tests/Unit/Index/Service/ResultIndexerTest.php (hunk 2 above)
Comment (hectoras) [reply tests/Unit/Index/Service/ResultIndexerTest.php]: Changed: bcd99bfe18478174878927a39ac61fbd0399d406 :ok_hand:

**Comment by hectoras** [comment]
- [x] Tests for extension-tao-advancedsearch fixed:

```
PHPUnit 8.5.23-8-g97d92d237 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.27 with Xdebug 3.1.2

......S..........................................                 49 / 49 (100%)

Time: 296 ms, Memory: 16.00 MB

There was 1 skipped test:

1) oat\taoAdvancedSearch\tests\Unit\Index\Report\IndexSummarizerTest::testProcess
TODO

/home/hector/PhpstormProjects/search-ADF-568/taoAdvancedSearch/tests/Unit/Index/Report/IndexSummarizerTest.php:40

OK, but incomplete, skipped, or risky tests!
Tests: 49, Assertions: 92, Skipped: 1.
```

**Review by shpran — APPROVED** [review]

**Comment by hectoras** [comment]
- [ ] Bump versions on a different MR (I've merged this one too early by mistake)

**Pull request merged**